src/lib/components/Error/WidgetsPropsValidator.tsx

```diff
@@ -0,0 +1,23 @@
+import { WidgetProps } from 'lib/components/Widget'
+import { IntegrationError } from 'lib/errors'
+import { PropsWithChildren, useEffect } from 'react'
+
+export default function WidgetsPropsValidator(props: PropsWithChildren<WidgetProps>) {
+  const { jsonRpcEndpoint, provider } = props
+
+  useEffect(() => {
+    if (!provider && !jsonRpcEndpoint) {
+      throw new IntegrationError('This widget requires a provider or jsonRpcEndpoint.')
+    }
+  }, [provider, jsonRpcEndpoint])
+
+  // size constraints
```

Comment:
nit: I don't think this comment is necessary because it's labeled by the error message. Same goes for the comments in the Swap validator.